tests/v2/validation/configmaps/configmaps_test.go

```diff
@@ -0,0 +1,113 @@
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+  "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+  "github.com/rancher/rancher/tests/framework/extensions/configmaps"
+)
+
+type SteveAPITestSuite struct {
+  suite.Suite
+  client            *rancher.Client
+  session           *session.Session
+  cluster           *management.Cluster
+}
+
+func (s *SteveAPITestSuite) TearDownSuite() {
+  s.session.Cleanup()
+}
+
+func (s *SteveAPITestSuite) SetupSuite() {
+  testSession := session.NewSession()
+  s.session = testSession
+
+  client, err := rancher.NewClient("", testSession)
+  require.NoError(s.T(), err)
+  s.client = client
+
+  log.Info("Getting cluster name from the config file and append cluster details in s")
+  clusterName := client.RancherConfig.ClusterName
+  require.NotEmptyf(s.T(), clusterName, "Cluster name to install should be set")
+  clusterID, err := clusters.GetClusterIDByName(s.client, clusterName)
+  require.NoError(s.T(), err, "Error getting cluster ID")
+  s.cluster, err = s.client.Management.Cluster.ByID(clusterID)
+  require.NoError(s.T(), err)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapCreate() {
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+  
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapNamespace := namespace
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, configmapNamespace, annotations, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was created")
+  configmapByID, err := configMapClient.ByID(configmapObj.ID)
+  require.NoError(s.T(), err)
+  assert.Contains(s.T(), configmapByID.JSONResp["id"], configmapName)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapUpdate(){
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+
+  log.Info(("Creating a configmap"))
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapNamespace := namespace
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, configmapNamespace, annotations, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Updating the configmap")
+  updatedConfigMapObj, err := updateConfigmapAnnotations(*configMapClient, configmapObj, updatedAnnotations)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was properly updated")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations, "anno2")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations["anno2"], "new automated annotation")
```

Comment:
Is it possible to not update the annotations during the initial creation of the configmap? Instead, we can create the configmap without annotations and then update it with the required annotations. We can define the list of newly added annotations as constants. During this step, we can then verify if the annotations have been successfully updated

Reply:
@anupama2501 do you mean to set the values of the map as constants? since maps can't be a constant in go from what I've found

Reply:
You can add the map as a var in the configmaps.go file. Similar to https://github.com/rancher/rancher/blob/release/v2.8/tests/v2/validation/rbac/rbac_psa_test.go#L85C1-L89C3